File: go/triangle/triangle.go
// This package determines if a triangle is equilateral, isosceles, or scalene.
package triangle

import "math"

type Kind int

const (
	NaT = iota // not a triangle
	Equ        // equilateral
	Iso        // isosceles
	Sca        // scalene
)

// Determines the triangle type.
func KindFromSides(a, b, c float64) Kind {
	if math.IsNaN(a+b+c) || math.IsInf(a+b+c, 0) ||
		a+b+c == 0 || a+b < c || a+c < b || b+c < a {
		return NaT
	} else if a == b && b == c {
		return Equ
	} else if a == b || a == c || b == c {
		return Iso
	}
	return Sca;
}

File: java/triangle/src/main/java/Triangle.java
class Triangle {

    double side1, side2, side3;

    Triangle(double side1, double side2, double side3) throws TriangleException {
        if(side1 <= 0 || side1 <= 0 || side3 <= 0 || side1 + side2 <= side3 || side1 + side3 <= side2 || side2 + side3 <= side1)
            throw new TriangleException();
        this.side1 = side1;
        this.side2 = side2;
        this.side3 = side3;
    }

    TriangleKind getKind() {
        if(side1 == side2 && side2 == side3) return TriangleKind.EQUILATERAL;
        if(side1 == side2 || side1 == side3 || side2 == side3) return TriangleKind.ISOSCELES;
        return TriangleKind.SCALENE;
    }

}
